hyperspy/_signals/eds.py

```diff
@@ -644,6 +644,10 @@ def get_lines_intensity(self,
                                                            FWHM_MnKa='auto')
             element, line = utils_eds._get_element_and_line(Xray_line)
             img = self.isig[window[0]:window[1]].integrate1D(-1)
+            if np.issubdtype(np.int32, np.integer):
```

Comment:
The thing is, while what you propose will indeed convert e.g. bool to float, preventing an error, I think that only in the case of int we should actually prevent an error, as if somebody tries to apply this to bool data there's something wrong going on.